src/SprykerEco/Zed/Unzer/Business/Payment/Finder/UnzerMarketplacePaymentUnzerCredentialsFinder.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Business\Payment\Finder;
+
+use Generated\Shared\Transfer\QuoteTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsConditionsTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsCriteriaTransfer;
+use Generated\Shared\Transfer\UnzerCredentialsTransfer;
+use Generated\Shared\Transfer\UnzerMarketplacePaymentCredentialsFinderCriteriaTransfer;
+use SprykerEco\Shared\Unzer\UnzerConstants;
+use SprykerEco\Zed\Unzer\Business\ApiAdapter\UnzerPaymentMethodsAdapterInterface;
+use SprykerEco\Zed\Unzer\Business\Exception\UnzerException;
+use SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface;
+
+class UnzerMarketplacePaymentUnzerCredentialsFinder implements UnzerMarketplacePaymentUnzerCredentialsFinderInterface
+{
+ /**
+  * @var \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface
+  */
+    protected UnzerReaderInterface $unzerReader;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\ApiAdapter\UnzerPaymentMethodsAdapterInterface
+     */
+    protected UnzerPaymentMethodsAdapterInterface $unzerPaymentMethodsAdapter;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface $unzerReader
+     * @param \SprykerEco\Zed\Unzer\Business\ApiAdapter\UnzerPaymentMethodsAdapterInterface $unzerPaymentMethodsAdapter
+     */
+    public function __construct(UnzerReaderInterface $unzerReader, UnzerPaymentMethodsAdapterInterface $unzerPaymentMethodsAdapter)
+    {
+        $this->unzerReader = $unzerReader;
+        $this->unzerPaymentMethodsAdapter = $unzerPaymentMethodsAdapter;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UnzerMarketplacePaymentCredentialsFinderCriteriaTransfer $unzerMarketplacePaymentCredentialsFinderCriteriaTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerCredentialsTransfer
+     */
+    public function findMarketplacePaymentUnzerCredentials(
+        UnzerMarketplacePaymentCredentialsFinderCriteriaTransfer $unzerMarketplacePaymentCredentialsFinderCriteriaTransfer
```

Comment:
I introduced own Criteria, because has nothing to do with fetching Unzer credentials. It's more kind of filter to check whether requirements are met or not.